app/frontend/src/features/messages/messagelist/InfiniteMessageLoader.tsx

```diff
@@ -66,11 +75,51 @@ export default function InfiniteMessageLoader({
     messageEl?.scrollIntoView();
   }, [isFetchingNextPage]);
 
-  //scroll to the bottom on page load
+  // Scroll to bottom on load
   useLayoutEffect(() => {
     if (!scrollRef.current) return;
     scrollRef.current.scroll(0, scrollRef.current.scrollHeight);
-  }, [scrollRef]);
+  }, []);
+
+  //**  Keep place or keep at bottom on window resize (i.e. keyboard popup on mobile (hopefully)) **//
+  // Save the current height outside of the eventlistener
+  const [currentClientHeight, setCurrentClientHeight] = useState(0);
+  useLayoutEffect(() => {
+    scrollRef.current && setCurrentClientHeight(scrollRef.current.clientHeight);
+  });
+
+  useLayoutEffect(() => {
+    const updateMessagePosition = () => {
+      if (!scrollRef.current) return;
+      const currentlyAtBottom =
+        scrollRef.current.scrollHeight <=
+        scrollRef.current.scrollTop + currentClientHeight;
+
+      if (currentlyAtBottom) {
+        // Send to bottom on window resize
+        scrollRef.current.scroll(0, scrollRef.current.scrollHeight);
+      }
+      setCurrentClientHeight(scrollRef.current.clientHeight);
+    };
+
+    if (!scrollRef.current) return;
+    window.addEventListener("resize", updateMessagePosition);
+    return () => window.removeEventListener("resize", updateMessagePosition);
+  }, [currentClientHeight]);
```

Comment:
You don't need the state for client height, since you can just access it from `scrollRef.current.clientHeight`. Then you don't need `currentClientHeight` as a dependency, and you can just add the event listener (it can be a normal effect rather than layout I think).

Reply:
```suggestion
   const updateMessagePosition = useCallback(() => {
    if (!scrollRef.current) return;
    const currentlyAtBottom =
      scrollRef.current.scrollHeight <=
      scrollRef.current.scrollTop + scrollRef.current.clientHeight;

    if (currentlyAtBottom) {
      // Send to bottom on window resize
      scrollRef.current.scroll(0, scrollRef.current.scrollHeight);
    }
  }, []);
    
  useEffect(() => {
    window.addEventListener("resize", updateMessagePosition);
    return () => window.removeEventListener("resize", updateMessagePosition);
  }, []);
```
(I haven't actually checked this code)

Reply:
This will also fix the dependency issues which is causing the build to fail (I think)

Reply:
The reason I had to save `clientHeight` is so we could compare the change in height. If we use scrollRef.current.clientHeight, then we wouldn't be looking at the height that it's changing from, so never passes `currentlyAtBottom`. 

FB messenger chat always sends you back to the bottom when you click the keyboard (no matter how far up you've scrolled), and asking my friends noone's felt like it was annoying, so we also could just do that (instead of me trying calculating whether they're at the bottom or not).

Makes it much neater overall

`
  const updateMessagePosition = useCallback(() => {
    if (!scrollRef.current) return;
    scrollRef.current.scroll(0, scrollRef.current.scrollHeight);
  }, []);

  useEffect(() => {
    window.addEventListener("resize", updateMessagePosition);
    return () => window.removeEventListener("resize", updateMessagePosition);
  });`